### ISSUE TITLE
[fusesoc] Fusesoc 0.2 breaks Nuvoton's release and freeze flow 

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#12891]
As discussed in more detail in #12769, Fusesoc 0.2 breaks the release and freeze flow from Nuvoton due to different temp folder naming for generated primitives. in order to unblock Nuvoton, Fusesoc has therefore been rolled back to 0.1 in #12874 for the time being - but this needs to be resolved so that we can continue rolling in updates.

**Comment by rswarbrick** [comment]
Great. @arnonsha: Please can you do a bit of digging to figure out what broke for you? I'm very happy to help with the fusesoc-digging side if you need it.

**Comment by arnonsha** [comment]
I can have a loik. Where can i find the source files of ot-fusesoc ?

**Comment by rswarbrick** [comment]
You can find it here: https://github.com/lowRISC/fusesoc

**Comment by arnonsha** [comment]
I grepped for "generated" in ot-0.1:
in ot-0.2 the "generate" function in edalizer.py uses /tmp instead of "generated" used in ot-0.1.

Hope this helps

**Comment by a-will** [comment]
> I grepped for "generated" in ot-0.1: in ot-0.2 the "generate" function in edalizer.py uses /tmp instead of "generated" used in ot-0.1.
> 
> Hope this helps

Probably due to [0469da452a68d44689bcb347708c7f4a775d3e41](https://github.com/lowRISC/fusesoc/commit/0469da452a68d44689bcb347708c7f4a775d3e41)

**Comment by vogelpi** [comment]
Update: I've now debugged our OT-specific fusesoc modifications. It turned out that during the last rebase, we forgot set some internal variable for cores generated by primgen (i.e., all the abstract cells) and we should have updated how outputs of generated cores are handled if the `--no-export` argument is provided. With the argument provided, the generator outputs weren't moved anywhere and we were relying on the temporary working directories under `/tmp` to stay around which is bad.

I've fixed these things, created a new tag and did a PR to switch to these versions. This should resolve this issue. I will wait for the green light from @arnonsha before merging this.
